### PR TITLE
doc: modem: documents LTE fallback mode

### PR DESCRIPTION
**Pull request by VeijoPesonen**
Makes fallback mode visible in our documentation.

### CODE REVIEW

**Comment by VeijoPesonen** [comment]
@charlieshao5189 Please review.

File: doc/nrf/libraries/modem/lte_lc.rst [hunk 1]
```diff
@@ -219,6 +219,16 @@ To enable modem sleep and TAU pre-warning notifications, enable the following op
 
 Additional configurations related to these features can be found in the API documentation for the link controller.
 
+Connection fallback mode
+========================
+It is possible to try to switch between LTE-M and NB-IoT after a certain time period if a connection has not been established.
+This comes useful in case connection to either network becomes unavailable.
```
Comment (umapraseeda) [inline doc/nrf/libraries/modem/lte_lc.rst:225]:
```suggestion
This is useful when the connection to either of these networks becomes unavailable.
```

File: doc/nrf/libraries/modem/lte_lc.rst [hunk 2]
```diff
@@ -219,6 +219,16 @@ To enable modem sleep and TAU pre-warning notifications, enable the following op
...
+Switching period between the network modes is configurable.
```
Comment (umapraseeda) [inline doc/nrf/libraries/modem/lte_lc.rst:226]:
```suggestion
You can also configure the switching period between the network modes.
```

File: doc/nrf/libraries/modem/lte_lc.rst [hunk 3]
```diff
@@ -219,6 +219,16 @@ To enable modem sleep and TAU pre-warning notifications, enable the following op
...
+If a connection cannot be established by using the fallback mode the library will report an error.
```
Comment (umapraseeda) [inline doc/nrf/libraries/modem/lte_lc.rst:227]:
```suggestion
If a connection cannot be established by using the fallback mode, the library reports an error.
You can use the following configuration options to configure the connection fallback mode:
```

**Comment by VeijoPesonen** [comment]
@umapraseeda I'll squash your changes and rebase. Takes a minute.

**Review by charlieshao5189 — APPROVED** [review]

**Comment by VeijoPesonen** [comment]
@umapraseeda we should be good in here.

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**